onnxruntime/core/providers/cpu/nn/string_concat.cc

```diff
@@ -0,0 +1,46 @@
+#include "string_concat.h"
+#include "core/providers/cpu/math/element_wise_ops.h"
+#include "core/common/common.h"
+
+namespace onnxruntime {
+ONNX_CPU_OPERATOR_KERNEL(
+    StringConcat,
+    20,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<std::string>()),
+    StringConcat);
+
+Status StringConcat::Compute(OpKernelContext* context) const {
+  ProcessBroadcastSpanFuncs broadcast_funcs{
+      [](BroadcastHelper& broadcast_helper) {
+        auto x = broadcast_helper.ScalarInput0<std::string>();
+        auto Y = broadcast_helper.SpanInput1<std::string>();
+        auto output = broadcast_helper.OutputSpan<std::string>();
+        std::transform(Y.begin(), Y.end(), output.begin(),
+                       [&x](const std::string& y) {
+                         return x + y;
```

Comment:
This  is not performant due to lots of re-allocations and copies.
I suggest doing broadcast separate, compute the length of the resulting string, pre-allocate space and append to it.

Reply:
I am not sure I understand the suggestion. I understand the issue of how best to concatenate the two strings into a target string (to avoid redundant temporaries/copies etc.). But I am unclear about your suggestion for doing something during broadcast, and something later, for parallelization.

Reply:
I would guess that @yuslepukhin would prefer to avoid any memory allocation when doing the actual string concatenation in parallel, but could you clarify this @yuslepukhin ?

Reply:
But the memory-allocation needs to happen at some point? We can minimize it (to one allocation), but cannot avoid it entirely, right?

Reply:
Disregard this comment.